lib/service/cfg.go

```diff
@@ -276,6 +276,29 @@ func (cfg *Config) ApplyToken(token string) (bool, error) {
 	return false, nil
 }
 
+// ApplyCAPins assigns the given CA pin(s), filtering out empty pins.
+func (cfg *Config) ApplyCAPins(caPins []string) error {
+	if len(caPins) == 0 {
+		return nil
+	}
+	filteredPins := make([]string, 0, len(caPins))
+	for _, pin := range caPins {
+		if pin != "" {
```

Comment:
Nit: I'd invert the condition to `if pin == "" { continue }` and unindent the block below.